File: 0274-h-index/0274-h-index.java
class Solution {
    public boolean helper(int[] a,int mid){
        int n=a.length;
        int count=0;
        for(int i=0;i<n;i++){
            if(a[i]>=mid) count++;
        }
        if(count>=mid) return true;
        else return false;
    }
    public int hIndex(int[] citations) {
        int n=citations.length;
        int low=0;
        int high=Integer.MIN_VALUE;

        for(int i=0;i<n;i++){
            high=Math.max(citations[i],high);
        }
        int ans=0;
        while(low<=high){
            int mid=(low+high)/2;
            if(helper(citations,mid)){
                ans=mid;
                low=mid+1;
            }
            else high=mid-1;
        }


        return ans;

    }
}

File: 0274-h-index/0274-h-index.cpp
class Solution {
public:
    bool helper(vector<int>& v,int x){
        int n=v.size();
        int count=0;
        for(int i=0;i<n;i++){
            if(v[i]>=x) count++;
        }
        if(count>=x) return true;
        return false;
    }
    int hIndex(vector<int>& citations) {
        int n=citations.size();
        int i=0;
        int j=-1;
        for(int i=0;i<n;i++){
            j=max(j,citations[i]);
        }
        int ans=-1;
        while(i<=j){
            int mid=(i+j)/2;
            if(helper(citations,mid)) {
                ans=max(ans,mid);
                i=mid+1;
                
            }
            else {
                j=mid-1;
               
            }
            

        }
        return ans;
    }
};